In [73]:
import pandas as pd

In [74]:
import os

print(os.listdir("/content"))

['.config', 'bc_card_out2020_03.txt', 'sample_data']


In [75]:
chunks = pd.read_csv(
    "/content/bc_card_out2020_03.txt",
    sep="\t",
    encoding="utf-8",
    chunksize=100000
)

데이터 불러오기

In [76]:
df = pd.concat(chunks, ignore_index=True)

In [77]:
df["TP_BUZ_NM"] = df["TP_BUZ_NM"].str.replace(" ", "", regex=False)

In [105]:
df["거주구분"] = "서울 비거주"

condition = df["CSTMR_MEGA_CTY_NM"] == "서울특별시"
df.loc[condition, "거주구분"] = "서울 거주"

In [106]:
print(df.shape)
df.head()

(1589494, 24)


,REG_YYMM,MEGA_CTY_NO,MEGA_CTY_NM,CTY_RGN_NO,CTY_RGN_NM,ADMI_CTY_NO,ADMI_CTY_NM,MAIN_BUZ_CODE,MAIN_BUZ_DESC,TP_GRP_NO,...,CSTMR_MEGA_CTY_NO,CSTMR_MEGA_CTY_NM,CSTMR_CTY_RGN_NO,CSTMR_CTY_RGN_NM,SEX_CTGO_CD,AGE_VAL,FLC,AMT,CNT,거주구분
0,202003,11,서울특별시,1168,강남구,11680750,수서동,30,생활,40,...,11,서울특별시,1168,강남구,2,20대,1,7927440,1089,서울 거주
1,202003,11,서울특별시,1129,성북구,11290660,길음1동,30,생활,70,...,11,서울특별시,1129,성북구,1,40대,4,274100,25,서울 거주
2,202003,11,서울특별시,1144,마포구,11440555,아현동,30,생활,40,...,11,서울특별시,1114,중구,1,60대 이상,5,34395725,808,서울 거주
3,202003,11,서울특별시,1126,중랑구,11260590,상봉2동,30,생활,40,...,11,서울특별시,1126,중랑구,2,20대,1,31860800,3699,서울 거주
4,202003,11,서울특별시,1168,강남구,11680630,대치4동,30,생활,62,...,41,경기도,4146,용인시,2,50대,4,2546487,45,서울 비거주


컬럼명을 알아보자

In [107]:
df.columns

Index(['REG_YYMM', 'MEGA_CTY_NO', 'MEGA_CTY_NM', 'CTY_RGN_NO', 'CTY_RGN_NM',
       'ADMI_CTY_NO', 'ADMI_CTY_NM', 'MAIN_BUZ_CODE', 'MAIN_BUZ_DESC',
       'TP_GRP_NO', 'TP_GRP_NM', 'TP_BUZ_NO', 'TP_BUZ_NM', 'CSTMR_GUBUN',
       'CSTMR_MEGA_CTY_NO', 'CSTMR_MEGA_CTY_NM', 'CSTMR_CTY_RGN_NO',
       'CSTMR_CTY_RGN_NM', 'SEX_CTGO_CD', 'AGE_VAL', 'FLC', 'AMT', 'CNT',
       '거주구분'],
      dtype='object')

REG_YYMM
= 기준 연월

MEGA_CTY_NM
= 카드를 사용한 시·도

CTY_RGN_NM
= 카드를 사용한 구

ADMI_CTY_NM
= 카드를 사용한 동

TP_BUZ_NM
= 세부 업종

CSTMR_MEGA_CTY_NM
= 고객이 사는 시·도

CSTMR_CTY_RGN_NM
= 고객이 사는 구

SEX_CTGO_CD
= 성별 코드

AMT
= 소비 금액

CNT
= 이용 건수



In [108]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1589494 entries, 0 to 1589493
Data columns (total 24 columns):
 #   Column             Non-Null Count    Dtype 
---  ------             --------------    ----- 
 0   REG_YYMM           1589494 non-null  int64 
 1   MEGA_CTY_NO        1589494 non-null  int64 
 2   MEGA_CTY_NM        1589494 non-null  object
 3   CTY_RGN_NO         1589494 non-null  int64 
 4   CTY_RGN_NM         1589494 non-null  object
 5   ADMI_CTY_NO        1589494 non-null  int64 
 6   ADMI_CTY_NM        1589494 non-null  object
 7   MAIN_BUZ_CODE      1589494 non-null  int64 
 8   MAIN_BUZ_DESC      1589494 non-null  object
 9   TP_GRP_NO          1589494 non-null  int64 
 10  TP_GRP_NM          1589494 non-null  object
 11  TP_BUZ_NO          1589494 non-null  int64 
 12  TP_BUZ_NM          1589494 non-null  object
 13  CSTMR_GUBUN        1589494 non-null  object
 14  CSTMR_MEGA_CTY_NO  1589494 non-null  int64 
 15  CSTMR_MEGA_CTY_NM  1589494 non-null  object
 16  

데이터 전처리 전에 상태 확인.

(빈칸이 있는지,
숫자인지 글자인지,
데이터가 몇 개 있는지.)

In [109]:
df.isnull().sum()

,0
REG_YYMM,0
MEGA_CTY_NO,0
MEGA_CTY_NM,0
CTY_RGN_NO,0
CTY_RGN_NM,0
ADMI_CTY_NO,0
ADMI_CTY_NM,0
MAIN_BUZ_CODE,0
MAIN_BUZ_DESC,0
TP_GRP_NO,0


각 컬럼의 빈칸 개수를 확인.

In [110]:
df["TP_BUZ_NM"].unique()

array(['편의점', '약국', '인터넷P/G', '생명보험', '세탁소', '일반한식', '의원', '서양음식', '가전제품',
       '일반서적', '택시', '슈퍼마켓', '인터넷종합Mall', '연쇄점', '공공요금', '기타음료식품',
       '기타숙박업', '대형할인점', '주점', '이동통신요금', '제과점', '유흥주점', '일식회집', '주차장',
       '정육점', '기타레져업', '악세사리', '상품권', '인터넷Mall', '사무서비스', '농,축협직영매장',
       '중국음식', '농축수산품', '스넥', '화물운송', '조세서비스', '미용원', '내의판매점', '기타잡화',
       '홈쇼핑', '가례서비스', '국산신차', '자동차정비', '병원', '항공사', '출판인쇄물', '화장품',
       '주방용구', '정장', '손해보험', '기타의료기관및기기', '애완동물', '유아원', '침구수예점', '고속버스',
       '문구용품', '종합병원', '볼링장', '구내매점(국가기관등)', '특급호텔', '안경', '화원', '기계공구',
       '스포츠레져용품', '의료용품', '피부미용실', '영화관', 'PG상품권', '정수기', '동물병원', '사우나',
       '전자상거래상품권', '한의원', '보습학원', '기타건강식', '치과의원', '골프경기장', '기타유통업',
       '컴퓨터', '소프트웨어', '노래방', '건강진단', '주유소', '부동산중개/임대', '기타교육',
       '기타용역서비스', '철도', '단란주점', '비료/농약/사료/종자', '스포츠의류', '위탁급식업',
       '학원(회원제형태)', '당구장', '기념품점', '종합용역', '아동의류', '기타주방용구', '기타사무용',
       '가정용품수리', '통신판매1', '기타회원제형태업소', '독서실', '학습지교육', '칵테일바', '완구점',
       '위성방송', '사무

TP_BUZ_NM 업종 칸에
어떤 종류의 값들이 들어 있는지 보기

In [111]:
df.duplicated().sum()

np.int64(0)

중복 데이터 확인. / 업종 값 확인. / 겹치지 않게 한 번씩만 보여주기.

In [112]:
df[df["TP_BUZ_NM"].str.contains("편", na=False)]["TP_BUZ_NM"].unique()

array(['편의점'], dtype=object)

띄어쓰기 수정

# **# 1. 서울시 거주 / 비거주 고객 수.**

In [113]:
df["거주구분"] = "서울 비거주"

In [114]:
condition = df["CSTMR_MEGA_CTY_NM"] == "서울특별시"

df.loc[condition, "거주구분"] = "서울 거주"

In [115]:
df[["CSTMR_MEGA_CTY_NM", "거주구분"]].head(10)

,CSTMR_MEGA_CTY_NM,거주구분
0,서울특별시,서울 거주
1,서울특별시,서울 거주
2,서울특별시,서울 거주
3,서울특별시,서울 거주
4,경기도,서울 비거주
5,서울특별시,서울 거주
6,서울특별시,서울 거주
7,서울특별시,서울 거주
8,서울특별시,서울 거주
9,서울특별시,서울 거주


In [116]:
df["거주구분"].value_counts()

,count
거주구분,
서울 거주,901072
서울 비거주,688422


In [117]:
df.groupby("거주구분")["CNT"].sum()

,CNT
거주구분,
서울 거주,58970629
서울 비거주,53351160


행 개수 기준

- 서울 거주: 901,072행
- 서울 비거주: 688,422행

카드 이용건수(CNT) 기준

- 서울 거주: 58,970,629건
- 서울 비거주: 53,351,160건

# **# 2. 총 소비액 구하기**

In [94]:
df.groupby("거주구분")["AMT"].sum()

,AMT
거주구분,
서울 거주,1385914569631
서울 비거주,1940899349900


# **# 3. 성별 소비액**

In [95]:
df["SEX_CTGO_CD"].value_counts()

,count
SEX_CTGO_CD,
1,865939
2,723555


성별코드 확인

In [96]:
df.groupby(["거주구분", "SEX_CTGO_CD"])["AMT"].sum()

거주구분    SEX_CTGO_CD
서울 거주   1               682678945342
        2               703235624289
서울 비거주  1              1015425463575
        2               925473886325
Name: AMT, dtype: int64

거주구분과 성별로 나눠 소비액 측정

서울 거주

- 성별코드 1: 682,678,945,342원
- 성별코드 2: 703,235,624,289원

서울 비거주

- 성별코드 1: 1,015,425,463,575원
- 성별코드 2: 925,473,886,325원

# **# 4. 편의점 소비정보 분석.**

(편의점 데이터만 꺼내기)

In [97]:
편의점 = df[df["TP_BUZ_NM"] == "편의점"]

In [98]:
print(편의점.shape)
편의점.head()

(180180, 24)


,REG_YYMM,MEGA_CTY_NO,MEGA_CTY_NM,CTY_RGN_NO,CTY_RGN_NM,ADMI_CTY_NO,ADMI_CTY_NM,MAIN_BUZ_CODE,MAIN_BUZ_DESC,TP_GRP_NO,...,CSTMR_MEGA_CTY_NO,CSTMR_MEGA_CTY_NM,CSTMR_CTY_RGN_NO,CSTMR_CTY_RGN_NM,SEX_CTGO_CD,AGE_VAL,FLC,AMT,CNT,거주구분
0,202003,11,서울특별시,1168,강남구,11680750,수서동,30,생활,40,...,11,서울특별시,1168,강남구,2,20대,1,7927440,1089,서울 거주
3,202003,11,서울특별시,1126,중랑구,11260590,상봉2동,30,생활,40,...,11,서울특별시,1126,중랑구,2,20대,1,31860800,3699,서울 거주
17,202003,11,서울특별시,1147,양천구,11470560,신월1동,30,생활,40,...,11,서울특별시,1147,양천구,1,30대,2,11396340,1671,서울 거주
19,202003,11,서울특별시,1156,영등포구,11560535,영등포동,30,생활,40,...,11,서울특별시,1153,구로구,1,20대,1,4959300,1000,서울 거주
42,202003,11,서울특별시,1168,강남구,11680640,역삼1동,30,생활,40,...,11,서울특별시,1168,강남구,1,30대,2,70223170,10085,서울 거주


전체 데이터에서 , 업종이 편의점인 줄만 골라서 *'편의점'*이라는 새 표에 넣기

# **#  5. 편의점 전체 소비액**

In [100]:
편의점_소비액 = 편의점["AMT"].sum()
print(f"{편의점_소비액:,}원")

79,987,167,291원


# **# 6. 강남구 편의점 소비액.**

In [101]:
강남구_편의점 = 편의점[
    편의점["CTY_RGN_NM"] == "강남구"
]

강남구_편의점_소비액 = 강남구_편의점["AMT"].sum()

print(f"{강남구_편의점_소비액:,}원")

8,170,947,461원


# **# 7. 서울시 거주/비거주 고객의 편의점 소비액**

In [102]:
편의점_거주별_소비액 = 편의점.groupby("거주구분")["AMT"].sum()

for 구분, 금액 in 편의점_거주별_소비액.items():
    print(구분, f"{금액:,}원")

서울 거주 66,513,977,146원
서울 비거주 13,473,190,145원


# **# 8. 거주지 소재 편의점 소비액**

고객이 사는 지역과 편의점이 있는 지역이 같은 소비만 고르기



*  고객 거주지: 서울특별시 강남구
*  편의점 위치: 서울특별시 강남구




In [103]:
거주지_편의점 = 편의점[
    (편의점["MEGA_CTY_NM"] == 편의점["CSTMR_MEGA_CTY_NM"]) &
    (편의점["CTY_RGN_NM"] == 편의점["CSTMR_CTY_RGN_NM"])
]

거주지_편의점_소비액 = 거주지_편의점["AMT"].sum()

print(f"{거주지_편의점_소비액:,}원")

44,184,614,834원


[코드 해석]

편의점["MEGA_CTY_NM"] == 편의점["CSTMR_MEGA_CTY_NM"]

카드를 쓴 시·도와 고객이 사는 시·도가 같은가?



---



편의점["CTY_RGN_NM"] == 편의점["CSTMR_CTY_RGN_NM"]

카드를 쓴 구와 고객이 사는 구가 같은가?



---


&

두 조건이 모두 맞는 행만 고르기.

# **# 9. 지역별 소비액**

In [104]:
거주지_편의점.groupby("CTY_RGN_NM")["AMT"].sum().sort_values(ascending=False)

,AMT
CTY_RGN_NM,
강남구,3102508044
관악구,2881385603
송파구,2748399782
강서구,2742806172
은평구,2094366779
마포구,2025802622
노원구,1978113730
강동구,1920253234
영등포구,1851803070


거주지와 같은 구에서 쓴 편의점 소비액을
구별로 묶고
*많은 순서대로 정렬*

# **# 10. 거주구분별 건당 소비금액**

In [118]:
건당_소비금액 = df.groupby("거주구분").agg(
    총소비액=("AMT", "sum"),
    이용건수=("CNT", "sum")
)

건당_소비금액["건당소비액"] = (
    건당_소비금액["총소비액"]
    / 건당_소비금액["이용건수"]
)

건당_소비금액

,총소비액,이용건수,건당소비액
거주구분,,,
서울 거주,1385914569631,58970629,23501.776955
서울 비거주,1940899349900,53351160,36379.702895




---



# **결론**

서울시 카드 소비 데이터를 분석한 결과, 서울 거주 고객은 서울 비거주
고객보다 카드 이용건수가 많았다. 반면 총소비액은 서울 비거주 고객이
더 높게 나타났다.


이용건수와 총소비액을 함께 비교한 결과, 서울 비거주 고객의 건당
소비금액이 서울 거주 고객보다 높게 나타났다. 따라서 서울 거주 고객은
상대적으로 자주 카드를 이용하고, 서울 비거주 고객은 한 번 이용할 때
더 큰 금액을 소비하는 경향이 있다고 해석할 수 있다.

편의점 소비액은 총 79,987,167,291원이었으며, 이 중 서울 거주 고객의
소비액은 66,513,977,146원으로 서울 비거주 고객보다 높게 나타났다.
이를 통해 서울 지역 편의점 소비는 주로 서울 거주 고객의 일상적인
생활 소비에서 발생한 것으로 볼 수 있다.

강남구 편의점 소비액은 8,170,947,461원이었으며, 고객의 거주지와
편의점 소재지가 같은 경우를 지역별로 분석한 결과 강남구의 소비액이
가장 높았다. 이를 통해 편의점 소비가 고객의 생활권과 밀접한 관련이
있음을 확인할 수 있었다.

(본 분석은 1,589,494행을 기준으로 분석하였다.)